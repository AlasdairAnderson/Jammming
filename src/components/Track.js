import React, { useState }  from 'react';

function Track(props) {
    const [trackImg, setTrackImg] = useState('');
    const [trackName, setTrackName] = useState('');
    const [albumName, setAlbumName] = useState('');
    const [artistName, setArtistName] = useState('');

    return(
        <div className="track">
            <img className="track-img" src={trackImg}/>
            <p className="track-name">{trackName}</p>
            <p className="album-name">{albumName}</p>
            <p classname="artist-name">{artistName}</p>
            <button alt="Add track to playlist">Add Track</button>
        </div>
    )
};

export default Track;