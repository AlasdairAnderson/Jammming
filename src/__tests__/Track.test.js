import React from 'react';
import { render, screen } from '@testing-library/react';
import Track from '../components/Track';

describe('<Track />', () => {
    const mockProps = {
        albumArt: 'https://www.google.com/url?sa=i&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FDecide_%2528album%2529&psig=AOvVaw3UFpAxfXEeNFlBL1mGOdPH&ust=1736004933558000&source=images&cd=vfe&opi=89978449&ved=0CBQQjRxqFwoTCICInL7w2YoDFQAAAAAdAAAAABAE',
        song: 'On and On',
        album: 'Go for it',
        artist: 'Djo'
    };

    it('renders the track image', () => {
        render(<Track {...mockProps} />);
        const imgElement = screen.getByRole('img');
        expect(imgElement).toBeInTheDocument();
        expect(imgElement).toHaveAttribute('src', mockProps.albumArt);
        expect(imgElement).toHaveClass('track-img');
    });

    it('renders the track name', () => {
        render(<Track {...mockProps} />);
        expect(screen.getByText(mockProps.song)).toBeInTheDocument();
        expect(screen.getByText(mockProps.song)).toHaveClass('track-name');
    });

    it('renders the album name', () => {
        render(<Track {...mockProps}/>);
        expect(screen.getByText(mockProps.album)).toBeInTheDocument();
        expect(screen.getByText(mockProps.album)).toHaveClass('album-name');
    });

    it('renders the artists name', () => {
        render(<Track {...mockProps}/>);
        expect(screen.getByText(mockProps.artist)).toBeInTheDocument();
        expect(screen.getByText(mockProps.artist)).toHaveClass('artist-name');
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(<Track {...mockProps}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should have an accessible image with an alt atttribute', () => {
        render(<Track {...mockProps}/>);
        const imgElement = screen.getByRole('img');
        expect(imgElement).toHaveAttribute('alt', expect.any(String));
    });

    it('renders default values when props are missing', () => {
        render(<Track />);
        expect(screen.queryByText(/Track name/i)).toBeNull();
    });

    it('does not render elements if props are missing', () => {
        render(<Track />);
        expect(screen.queryByText(/Text Song/i)).not.toBeInTheDocument();
    });
})