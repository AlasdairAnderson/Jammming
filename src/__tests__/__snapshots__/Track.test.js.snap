// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> matches the snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Album art from On and On"
    class="track-img"
    src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FDecide_%2528album%2529&psig=AOvVaw3UFpAxfXEeNFlBL1mGOdPH&ust=1736004933558000&source=images&cd=vfe&opi=89978449&ved=0CBQQjRxqFwoTCICInL7w2YoDFQAAAAAdAAAAABAE"
  />
  <div
    class="track-info"
  >
    <p
      class="track-name"
    >
      On and On
    </p>
    <p
      class="album-name"
    >
      Go for it
    </p>
    <p
      class="artist-name"
    >
      Djo
    </p>
  </div>
</DocumentFragment>
`;
